group_project_v2/group_project.py

```diff
@@ -458,8 +560,124 @@ def dashboard_view(self, context):
 
     @outsider_disallowed_protected_view
     def dashboard_detail_view(self, context):
-        # TODO: implement detail view
-        pass
+        fragment = Fragment()
+
+        children_context = context.copy()
+
+        target_workgroups = context.get(DashboardRootXBlockMixin.TARGET_WORKGROUPS)
+        target_users = context.get(DashboardRootXBlockMixin.TARGET_STUDENTS)
+
+        stages = []
+        stage_stats = {}
+        for stage in self.stages:
+            if not stage.is_graded_stage:
+                continue
+            stage_fragment = stage.render('dashboard_detail_view', children_context)
+            stage_fragment.add_frag_resources(fragment)
+            stages.append({"id": stage.id, 'content': stage_fragment.content})
+            stage_stats[stage.id] = self._get_stage_completion_details(stage, target_workgroups, target_users)
+
+        groups_data = self._build_groups_data(target_workgroups, stage_stats)
+
+        render_context = {
+            'activity': self, 'stages': stages, 'stages_count': len(stages), 'groups': groups_data,
```

Comment:
@e-kolpakov Nit: Line breaks after every item would make the contents of this dict easier to read.